design.md

```diff
@@ -0,0 +1,128 @@
+# MAJOR GOAL:
+have most of this stuff as uncoupled as possible. entire reason for doing this
+is so that it's easy to extend, and also easy for those unfamilar with the script
+to jump in and extend.
+
+# core
+creates FreeBSDSyscall object, which should instantiate itself with everything
+it needs to
+then, just run down the modules
+    (e.g., like gen_syscalls_h)
+and achieve the same output
+    init_sysent.lua
+    syscall_h.lua
+    syscalls.lua
+    syscalls_mk.lua
+    systrace_args.lua
+
+### ALSO needs to:
+have a method of opting out of complex file generation.
+shouldn't really be a hastle though.
+probably do what it needs to do by default, and pass inclusion/exclusion args
+    e.g., ./makesyscalls --include=init_sysent,syscall  # only does init_sysent.h and syscall.c
+          ./makesyscalls --exclude=systrace_args        # does everything except systrace_args.h
+    also a great place to provide man page and further online documentation
```

Comment:
I'd go farther and say `that` makesyscalls should 100% disappear.
It was one big monolithic program because it was just easier back in the day to parse once and output many things. Now, I don't think that makes much sense and you'd waste a lot of time on include/exclude that I think we don't really need anymore.

Reply:
IMO splitting generation of the various files out into their own front-end and targets off of `sysent` would be a good test of the lib'ified model, making sure it's as ergonomic as we'd like for these different exact needs to use.

Reply:
This is somewhat of a crossroads. I'm leaning more towards splitting it off for the testing aspect, but I want to stay focused on what you guys need, so I'll put a pin in that for now.